### PR TITLE
Relocate extension files from Omise/Payment out to root dir

### PR DESCRIPTION
**Pull request by guzzilar**
#### 1. Objective

Because Magento 2 allows merchant to install an extension through `composer` command. 

To support that, we should register Omise-Magento repository to [Packagist](https://packagist.org) package manager. Which is, requires `composer.json` to be located in a root directory.

So, we need to relocate our composer.json & all files out from `Omise/Payment` folder to root directory.

**Related information**:
Related issue(s): 🙅
Related ticket(s): _T2080_

#### 2. Description of change

Because this PR is about relocate files, which is it's gonna make ton of changes and hard to review.
So, I've captured commands that I did for this PR, please check below.

screen shot 2559-12-15 at 3 28 23 am

#### 3. Quality assurance

Not really, but you could see that all of files in `Omise/Payment` folder was moved to root dir.

#### 4. Impact of the change

This PR is just a one step to adjust Omise-Magento to be able to upload to Magento extension store which is it'd not finish within itself.

To clarify the state, this PR would break something and couldn't live without following PRs (that I would submit to review soon).

So, that is the reason I wanted to merge this PR to `2-0-dev` instead of some main branch.

The next PR would be something solve a real issue that blocks us to submit plugin to Magento's store.

#### 5. Priority of change

Normal

#### 6. Additional Notes

No.

### CODE REVIEW

**Comment by nimid** [comment]
@guzzilar Can you explain further?

- In the section 3. Quality assurance, I have read the explaination of Plugin Contributing Template. What is quality assurance in this pull request?

- "**this PR would break something**", What, which part or which step this pull request break?

- "couldn't live without **following PRs**", Which branch you plan for `following PRs`, new branchs?

- I found a branch, `2-0-dev-adjust-for-magento-marketplace`, what you intend to do with this branch?

- "instead of **some main branch**", which branch you refer?

- What is "a real issue"?

- What do you want the reviewer to review in this pull request?

**Comment by guzzilar** [comment]
TL;DR;
Please skip this PR, I'll reopen & update some PR name & work flow to make it more easy to follow with. And let me ping you again when I finish.

Thanks 👍 

---

> In the section 3. Quality assurance, I have read the explaination of Plugin Contributing Template. What is quality assurance in this pull request?

This PR is about relocated file from one place to another place, so there is no specific case to test.
Instead, I should update "4. Impact of the change" topic.

...

> "this PR would break something", What, which part or which step this pull request break?

My fault, misunderstood myself. I'll update "4. Impact of the change" topic.

...

> "couldn't live without following PRs", Which branch you plan for following PRs, new branchs?

The same as above answer, this branch just relocated file from sub-dir to root dir which is it would impact to only installation part.

So, I'll update "4. Impact of the change" topic. (please check below) 

But for answer your question "Which branch you plan for following PRs, new branchs?"
I planed to submit another PR with "2-0-dev-adjust-for-magento-marketplace" branch.

This PR is just one part of "adjust-for-magento-marketplace" plan, but to make PR easy to review, I decided to split this changes to this PR.

...

> I found a branch, 2-0-dev-adjust-for-magento-marketplace, what you intend to do with this branch?

As the branch name says, adjust Omise-Magento plugin to make it conform with Magento's extension store conditions. (which is, I would explain in another PR when I finish that branch.)

...

> "instead of some main branch", which branch you refer?

`2-1-0` (since that branch is a main branch for Magento 2.x series right now.)

...

> What is "a real issue"?

i.e. can't submit to Magento's extension store because Omise-Magento is not support namespace right now (in Gateway/Http/Lib/omise-php)

For others I couldn't answer right now, still figure out & re-submit extension.

...

> What do you want the reviewer to review in this pull request?

Just wants reviewer notice this change, in case that maybe reviewer would have some question.

---


Btw,
To make it more clear & explicit, I would close this PR & re-submit this branch (`2-0-dev-adjust-file-location`) to `2-0-dev-adjust-for-magento-marketplace` instead.

So, my plan for this issue (to adjust/update plugin to match with Magento extension submission's guides), I would.
1. rename `2-0-dev-adjust-for-magento-marketplace` to `2-0-update-code-conform-with-magento-extension-guide`
2. rename `2-0-dev-adjust-file-location` to `2-0-adjust-file-location`
3. open new PR from `2-0-adjust-file-location` to `2-0-update-code-conform-with-magento-extension-guide`
4. if everything finish, I would open PR from `2-0-update-code-conform-with-magento-extension-guide` to `2-1-0`

for branches flow, it's gonna be something like below.
```
• `2-0-adjust-file-location` ----------->  • `2-0-update-code-conform-with-magento-extension-guide` ---------> `2-1-0`
                                        /   /
• `2-0-update-omise-php` _____________ /   /
                                          /
• `2-0-blah` ____________________________/
```

Btw, nice catch! 👍 @nimid

**Comment by guzzilar** [comment]
note. the following branches was changed to new name.
1. **2-0-dev-adjust-for-magento-marketplace**
👇
`2-0-update-code-to-conform-with-magento-extension-guide`

2. **2-0-dev-adjust-file-location**
👇 
`2-0-relocate-files-to-root-dir`

3. **2-0-dev-adjust-for-magento-marketplace**
👇 
`2-0-update-composer-spec`
